testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/IdentityProviderTest.java

```diff
@@ -1071,8 +1074,22 @@ public void testSamlExportSignatureOn() throws URISyntaxException, IOException,
         response.close();
 
         Document document = DocumentUtil.getDocument(body);
+        System.out.println(DocumentUtil.getNodeAsString(document));
 
         Element signatureElement = DocumentUtil.getDirectChildElement(document.getDocumentElement(), XMLDSIG_NSURI.get(), "Signature");
-        Assert.assertNotNull(signatureElement);
+        Assert.assertNotNull("Signature not null", signatureElement);
+
+        Element keyInfoElement = DocumentUtil.getDirectChildElement(signatureElement, XMLDSIG_NSURI.get(), "KeyInfo");
+        Assert.assertNotNull("KeyInfo not null", keyInfoElement);
+
+        Element x509DataElement = DocumentUtil.getDirectChildElement(keyInfoElement, XMLDSIG_NSURI.get(), "X509Data");
+        Assert.assertNotNull("X509Data not null", x509DataElement);
+
+        Element x509CertificateElement = DocumentUtil.getDirectChildElement(x509DataElement, XMLDSIG_NSURI.get(), "X509Certificate");
+        Assert.assertNotNull("X509Certificate not null", x509CertificateElement);
+
+        System.out.println(DocumentUtil.getNodeAsString(keyInfoElement));
+        Element keyNameElement = DocumentUtil.getDirectChildElement(keyInfoElement, XMLDSIG_NSURI.get(), "KeyName");
+        Assert.assertNotNull("KeyName not null", keyNameElement);
```

Comment:
Can you please try to verify the signature too? I know it was not done before but it's better if we it now, it would be something like the following:

```java
        String keyId = realm.keys().getKeyMetadata().getActive().get(Constants.DEFAULT_SIGNATURE_ALGORITHM);
        String certificate = realm.keys().getKeyMetadata().getKeys().stream().filter(k -> keyId.equals(k.getKid())).findFirst().get().getCertificate();
        X509Certificate x509cert = XMLSignatureUtil.getX509CertificateFromKeyInfoString(certificate);
        SamlProtocolUtils.verifyDocumentSignature(document, new HardcodedKeyLocator(x509cert.getPublicKey()));
```

If you want you can also use those values to ensure the KeyInfo is containing those keyId and certificate values.

Reply:
Latest version of the tests now validates the signature, and the cert subject DN. The latter should be more in line with the rest of the test, I think, because it actually validates a certificate rather than a key property.